Sources/SwiftDocC/Utility/Graphs/DirectedGraph+Cycles.swift

```diff
@@ -0,0 +1,63 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+extension DirectedGraph {
+    /// Returns the first cycle in the graph encountered through breadth first traversal from a given starting point
+    ///
+    /// The cycle starts at the earliest repeated node and ends with the node that links back to the starting point.
+    ///
+    /// - Note: A cycle, if found, is guaranteed to contain at least one node.
+    func firstCycle(from startingPoint: Node) -> Path? {
```

Comment:
Fun coincidence; when I was looking at other similar debugging API we have I noticed that `TopicGraph.dump()` doesn't handle cycles so I added a call to `firstCycle(from:)` there to exit with a more descriptive error message instead of recursing infinitely. That will hopefully help someone who uses it for debugging and encounters a cycle.